Components/ProductToShop.php

```diff
@@ -949,74 +664,48 @@ public function hasFieldChanged($field, ProductModel $model, DetailModel $detail
     }
 
     /**
-     * Helper method to determine if a given $fields may/must be updated.
-     * This method will check for the model->id in order to determine, if it is a new entity. Therefore
-     * this method cannot be used after the model in question was already flushed.
-     *
-     * @param $field
-     * @param $model ProductModel
-     * @param $attribute ConnectAttribute
-     * @throws \RuntimeException
-     * @return bool|null
+     * @param $updateFields
+     * @param $model
+     * @param $detailAttribute
+     * @param Product $product
      */
-    public function isFieldUpdateAllowed($field, ProductModel $model, ConnectAttribute $attribute)
+    private function setPropertiesForNewProducts($updateFields, $model, $detailAttribute, Product $product)
```

Comment:
again type hints. ```$updateFields``` is array for example. ```$model``` & ```$detailAttribute``` are objects.
after that fix dochint